cmake/extensions.cmake

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# cddl_source(<target> <cddl_file> ENTRY_TYPES <entry_types> [VERBOSE])
+#
+# Add generated code to the project for decoding CBOR.
+#
+# <cddl_file> is a file with CDDL data describing the expected CBOR data.
+#
+# <entry_types> is a list of the types defined in the CDDL file for which there
+#               should be an exposed decoding function.
+#
+# Specify VERBOSE to add printing to the generated code. Check kconfig to add
+# printing in the low level CBOR decoding code.
+#
+# The result of the function is that a library is added to the project which
+# contains the generated decoding code. The code is generated at build time using
+# a script, and generates a c file and an h file with names derived from the
+# name of the <cddl_file>.
+function(cddl_source target cddl_file)
+  if(IS_ABSOLUTE ${cddl_file})
+    set(cddl_path ${cddl_file})
+  else()
+    set(cddl_path ${CMAKE_CURRENT_SOURCE_DIR}/${cddl_file})
+  endif()
+  get_filename_component(name ${cddl_path} NAME_WE)
+  set(c_file ${PROJECT_BINARY_DIR}/${name}.c)
+  set(h_file_dir ${PROJECT_BINARY_DIR}/zephyr/include/generated)
+  set(h_file ${h_file_dir}/${name}.h)
+
+  cmake_parse_arguments(CDDL "VERBOSE" "" "ENTRY_TYPES" ${ARGN})
+  if (${CDDL_VERBOSE})
+    set(verbose_flag "-v")
```

Comment:
Basically, if parent scope has defined the variable `verbose_flag`, then it will already be set even if `cddl_source` is called without `VERBOSE`.

And as `verbose_flag` is quite generic name, this could likely happen.

Reply:
Good point, how do I fix that?

    else()
      set(verbose_flag "")

?

Reply:
that's one option, or you can simply do:
```
set(verbose_flag)
if (${CDDL_VERBOSE})
    set(verbose_flag "-v")
endif()
```
depending on the preference.
There is also a third way, using a generator expression, then you avoid all the `if () .... else()`, see: https://github.com/oyvindronningstad/cddl_gen/pull/1/files#r374641345

Reply:
Cool, thanks.